packages/vite/src/node/optimizer/scan.ts

```diff
@@ -294,8 +294,23 @@ async function prepareEsbuildScanner(
     const { tsconfig } = await loadTsconfigJsonForFile(
       path.join(environment.config.root, '_dummy.js'),
     )
-    if (tsconfig.compilerOptions?.experimentalDecorators) {
-      tsconfigRaw = { compilerOptions: { experimentalDecorators: true } }
+    if (
+      tsconfig.compilerOptions?.experimentalDecorators ||
+      tsconfig.compilerOptions?.jsx ||
+      tsconfig.compilerOptions?.jsxFactory ||
+      tsconfig.compilerOptions?.jsxFragmentFactory ||
+      tsconfig.compilerOptions?.jsxImportSource
+    ) {
+      tsconfigRaw = {
+        compilerOptions: {
+          experimentalDecorators:
+            tsconfig.compilerOptions?.experimentalDecorators,
+          jsx: tsconfig.compilerOptions?.jsx,
+          jsxFactory: tsconfig.compilerOptions?.jsxFactory,
+          jsxFragmentFactory: tsconfig.compilerOptions?.jsxFragmentFactory,
+          jsxImportSource: tsconfig.compilerOptions?.jsxImportSource,
+        },
+      }
     }
   }
```

Comment:
I think we need
https://github.com/vitejs/vite/blob/f0880df0eab077e61abb533c4edde6538bd729f9/packages/vite/src/node/plugins/esbuild.ts#L167-L174
here ([esbuild try](https://esbuild.github.io/try/#YgAwLjI1LjgALS1qc3g9YXV0b21hdGljIC0tdHNjb25maWctcmF3PSd7ImNvbXBpbGVyT3B0aW9ucyI6eyJqc3giOiAicmVhY3QifX0nAGUAZW50cnkuanN4AGNvbnNvbGUubG9nKDxkaXYgLz4p)).